lib-multisrc/greenshit/src/eu/kanade/tachiyomi/multisrc/greenshit/GreenShitDto.kt

```diff
@@ -1,240 +1,259 @@
-            val imageUrl = when {
-                image.isWordPressContent() -> {
-                    CDN_URL.toHttpUrl().newBuilder()
-                        .addPathSegments("wp-content/uploads/WP-manga/data")
-                        .addPathSegments(image.src.toPathSegment())
-                        .build()
-                }
-                else -> {
-                    "$CDN_URL/scans/${dto.manga.scanId}/obras/${dto.manga.id}/capitulos/$chapter/${image.src}"
-                        .toHttpUrl()
-                }
-            }
-            Page(index, imageUrl = imageUrl.toString())
-        }
-    }
-
-    private fun Float.isNotInteger(): Boolean = toInt() < this
-
-    private fun String.createSlug(): String {
-        return Normalizer.normalize(this, Normalizer.Form.NFD)
-            .trim()
-            .replace("\\p{InCombiningDiacriticalMarks}+".toRegex(), "")
-            .replace("\\p{Punct}".toRegex(), "")
-            .replace("\\s+".toRegex(), "-")
-            .lowercase()
-    }
-
     companion object {
-        @SuppressLint("SimpleDateFormat")
-        val dateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.ROOT)
-        val CHAPTER_NUMBER_REGEX = """\d+(\.\d+)?""".toRegex()
+        val DATE_FORMAT = SimpleDateFormat("yyyy-MM-dd", Locale.ROOT)
     }
 }
 
-@Serializable
-class TokenDto(
-    @SerialName("token")
-    val value: String,
-)
+fun ResultDto<List<MangaDto>>.toSMangaList(cdnUrl: String, useWidth: Boolean, includeSlug: Boolean = true, defaultScanId: Int? = null): List<SManga> =
+    results.filter { it.type != "TEXTO" }.map { it.toSManga(cdnUrl, useWidth, includeSlug, defaultScanId) }
 
 @Serializable
 class MangaDto(
-    @SerialName("obr_id")
+    @JsonNames("obr_id", "id")
     val id: Int,
-    @SerialName("obr_descricao")
-    val description: String?,
-    @SerialName("obr_imagem")
-    val thumbnail: String?,
-    @SerialName("obr_nome")
-    val name: String,
-    @SerialName("obr_slug")
-    var slug: String?,
-    @SerialName("status")
-    val status: MangaStatus,
+    @JsonNames("obr_nome", "name")
+    val name: String? = null,
+    @JsonNames("obr_slug", "slug")
+    var slug: String? = null,
+    @JsonNames("obr_imagem", "image", "thumbnail")
+    val thumbnail: String? = null,
+    @JsonNames("obr_descricao", "description")
+    val description: String? = null,
     @SerialName("scan_id")
     val scanId: Int,
-    @SerialName("tags")
-    val genres: List<Genre>,
+    @JsonNames("status", "obr_status")
+    val status: MangaStatus? = null,
+    @JsonNames("tags", "genres")
+    val genres: List<Genre> = emptyList(),
+    @JsonNames("capitulos", "chapters")
+    val chapters: List<ChapterDto>? = null,
+    val type: String? = null,
 ) {
+    fun toSManga(cdnUrl: String, useWidth: Boolean, includeSlug: Boolean = false, defaultScanId: Int? = null) = SManga.create().apply {
+        title = name ?: throw IllegalArgumentException("Name is required for title")
```

Comment:
You don't have to be this explicit. You can just:
```suggestion
        title = name!!
```

or if every `MangaDto` you deserialize will eventually call `MangaDto.toSManga()` then you can move the constraint to the field itself.

```diff
-val name: String? = null,
+val name: String = null,
```

Comment applies to the slug as well.